### PR TITLE
chore(website): update the roadmap and remove changelog item

### PR DESCRIPTION
**Pull request by SiTaggart**
Update the roadmap page with new version numbering system.

Hide the changelog item from the sidebar for now

### CODE REVIEW

File: packages/paste-website/src/pages/roadmap/index.mdx [hunk 1]
```diff
@@ -32,57 +32,7 @@ export const pageQuery = graphql`
...
-  <Anchor href="https://twilio-labs.github.io/paste/?path=/story/overview-icons--list">
-    full icon list on Storybook
-  </Anchor>.
-
----
-
-## Release 0.6
+## Release 2020.02.25
 
 Release date: February 25, 2020
```
Comment (serifluous) [inline packages/paste-website/src/pages/roadmap/index.mdx]:
```
**Design releases**

- Design more form field components (Select, Checkbox and Checkbox group, and Radio and Radio Group). 📝
- Design a Form composition component. 📝
- Design a Toast component. 🍞

**Engineering releases**

TBD

**Other**

- Add an icon list to the [icon system guidelines](/icon-system) with more design guidelines. Right now, you can access our  
  <Anchor href="https://twilio-labs.github.io/paste/?path=/story/overview-icons--list">
    full icon list on Storybook
  </Anchor>.
- Make this website responsive 📱
```

File: packages/paste-website/src/pages/roadmap/index.mdx [hunk 2]
```diff
@@ -103,7 +53,7 @@ Release date: February 25, 2020
 
 ---
 
-## Release 0.7
+## Release 2020.04.07
 
 Release date: April 7, 2020
```
Comment (serifluous) [inline packages/paste-website/src/pages/roadmap/index.mdx]:
```
**Design releases**

- Design Overlay components (Modal, Alert dialog, and Overlay). 🕴
- Design a Tab component.
- Design a Table component.
- Define our icon-naming schema.

**Engineering releases**

TBD

**Other**

TBD
```

File: packages/paste-website/src/pages/roadmap/index.mdx (hunk 1 above)
Comment (SiTaggart) [reply packages/paste-website/src/pages/roadmap/index.mdx]: > - Make this website responsive 📱
Yeah, right 😛

Comment (serifluous) [reply packages/paste-website/src/pages/roadmap/index.mdx]: idk drop a grid on it??

ok fine, you're right. Take it out.

Comment (SiTaggart) [reply packages/paste-website/src/pages/roadmap/index.mdx]: removed

**Pull request merged**